translate/storage/jsonl10n.py

```diff
@@ -232,6 +237,9 @@ def getkey(self):
     def getvalue(self):
         ret = self.converttarget()
         for k in reversed(self.getkey()):
+            if '[' in k and k[-1] == ']':
+                k = k.split('[')[0]
```

Comment:
Anything odd going to happen if it's an empty array like `[]`?

Reply:
Well the keys are ignored anyway, with assumption that the list is constructed in order by units, so it will work just fine.

Yes, it can result in storing something else than was parsed, but that's how the JSON storage currently works (see https://github.com/translate/translate/issues/3819 and https://github.com/translate/translate/pull/3835).

Reply:
IOW: It will certainly break in some cases, but I believe it works better than current code. Some solution like outlined in #3835 is needed to make it work reliably in all corner cases.